static/js/activity.js

```diff
@@ -320,23 +321,86 @@ exports.insert_user_into_list = function (user_id) {
     compose_fade.update_one_user_row(elt);
 };
 
+function get_users(search_term) {
+    var all_persons = [];
+    if (search_term.indexOf('') !== -1) {
+        all_persons = presence.get_user_ids();
+    } else {
+        all_persons = people.get_all_persons_ids();
+    }
+    return all_persons;
+}
+
+exports.create_filter_function = function (search_terms) {
+    return function (item) {
+        var person = people.get_person_from_user_id(item);
+
+        // Get person object (and ignore errors)
+        if (!person || !person.full_name) {
+            return;
+        }
+
+        if (people.is_current_user(person.email)) {
+            return false;
+        }
+
+        var match = _.any(search_terms, function (search_term) {
+            return people.person_matches_query(person, search_term);
+        });
+
+        if (match) {
+            return true;
+        }
+        return false;
+    };
+};
+
+exports.retrieve_sidebar_users_ids = function (search_terms) {
+    var users = get_users(search_terms);
+    users = sort_users(users);
+
+    return users;
+};
+
+exports.process_search_terms = function () {
+    var user_list = $(".user-list-filter");
+    var search_term = user_list.expectOne().val().trim();
+    var search_terms = search_term.toLowerCase().split(",");
+    search_terms = _.map(search_terms, function (s) {
+        return s.trim();
+    });
+    return search_terms;
+};
+
+exports.row_modifier = function (item) {
+    var person_data = info_for(item);
+    var row = templates.render('user_presence_row', person_data);
```

Comment:
The function `templates.render()` returns a string, which is almost always a chunk of HTML.  Then `$(html)` makes a jQuery container for that piece of DOM.

jQuery makes is so that `$($($(html))) == $(html)`, but there is a difference between `$(html)` and `html`.